sim/dex.ts

```diff
@@ -363,14 +363,29 @@ export class ModdedDex {
 		}
 		if (!this.data.Pokedex.hasOwnProperty(id)) {
 			let aliasTo = '';
-			if (id.startsWith('mega') && this.data.Pokedex[id.slice(4) + 'mega']) {
-				aliasTo = id.slice(4) + 'mega';
-			} else if (id.startsWith('m') && this.data.Pokedex[id.slice(1) + 'mega']) {
-				aliasTo = id.slice(1) + 'mega';
-			} else if (id.startsWith('primal') && this.data.Pokedex[id.slice(6) + 'primal']) {
-				aliasTo = id.slice(6) + 'primal';
-			} else if (id.startsWith('p') && this.data.Pokedex[id.slice(1) + 'primal']) {
-				aliasTo = id.slice(1) + 'primal';
+			const formeNames: {[k: string]: string} = {
+				a: 'alola',
+				alolan: 'alola',
+				g: 'galar',
+				galarian: 'galar',
+				gigantamax: 'gmax',
+				gmax: 'gmax',
+				m: 'mega',
+				mega: 'mega',
+				p: 'primal',
+				primal: 'primal',
+			};
+			for (const forme in formeNames) {
+				if (id.startsWith(forme)) {
+					let pokeName = id.slice(forme.length);
```

Comment:
the errors are because you dont have `pokeName` declared anywhere outside of this if statement, declare it above the if statement as
```typescript
let pokeName = '' as string;
```

Reply:
ty i missed that